### PR TITLE
Updates to net-tools sample test script

### PR DESCRIPTION
**Pull request by pfl**
A few updates to scripts/net/run-sample-tests.sh script enabling overlays for the Zephyr build process and running both UDP and TCP echo tests.

### CODE REVIEW

File: scripts/net/run-sample-tests.sh [hunk 1]
```diff
@@ -394,6 +404,17 @@ do
 	    fi
 	    configuration=stop_only
 	    ;;
+
+	--overlay)
+	    shift
+	    if [ -n "$zephyr_overlay" ]
```
Comment (jukkar) [inline scripts/net/run-sample-tests.sh]: Can't we just say `zephyr_overlay="$zephyr_overlay $1"` here? Is there some shell that does not accept this?

File: scripts/net/run-sample-tests.sh [hunk 2]
```diff
@@ -164,7 +164,7 @@ start_zephyr ()
     fi
```
Comment (jukkar) [inline scripts/net/run-sample-tests.sh]: Why do we need / want to use 64-bit native_posix? That info would be nice to have in commit msg.

File: scripts/net/run-sample-tests.sh (hunk 1 above)
Comment (pfl) [reply scripts/net/run-sample-tests.sh]: We get an extra space into that variable in such a case, but that should mean harm.

File: scripts/net/run-sample-tests.sh (hunk 2 above)
Comment (pfl) [reply scripts/net/run-sample-tests.sh]: Actually, we don't. Let's skip this patch.

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**